src/main/java/lv/ctco/javaschool/goal/boundary/GoalApi.java

```diff
@@ -204,5 +203,22 @@ public UserDto getUserById(@PathParam("id") Long id) {
             throw new InvalidUserException();
         }
     }
+
+    @POST
+    @Path("/mygoals/{id}")
+    public void setStatusAchieved(@PathParam("id") Long goalId) {
+        User user = userStore.getCurrentUser();
+        Optional<Goal> goal = goalStore.getGoalById(goalId);
+        if (goal.isPresent()) {
+            Goal g = goal.get();
+            if (g.getUser() == user) {
```

Comment:
Only primitives, `String`s and `Enum`s may be compared this way. Never use `==` to compare objects unless you are explicitly comparing references to the same object. Use `equals()` (making sure the class implements both `equals()` and `hashCode()`), or compare user IDs.

Reply:
It may be possible that they are the same object due to entity persistence framework (Hibernate), however we cannot rely on that.

Reply:
added equals() and hashCode()